packages/backend/src/api/controllers/ActivityController.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ActivityChartPoint,
+  ApiActivity,
+  ProjectId,
+  UnixTime,
+} from '@l2beat/types'
+
+import { Config } from '../../config'
+import { RpcTransactionCountRepository } from '../../peripherals/database/RpcTransactionCountRepository'
+import { StarkexTransactionCountRepository } from '../../peripherals/database/StarkexTransactionCountRepository'
+
+type ProjectsCounts = {
+  projectId: ProjectId
+  counts: { timestamp: UnixTime; count: number }[]
+}[]
+
+type TransactionApiProjects = Pick<
+  Config['projects'][number],
+  'projectId' | 'transactionApi'
+>[]
+
+export class ActivityController {
+  constructor(
+    private projects: TransactionApiProjects,
+    private rpcRepository: RpcTransactionCountRepository,
+    private starkexRepository: StarkexTransactionCountRepository,
+  ) {}
+
+  async getTransactionActivity(): Promise<ApiActivity> {
+    const projectsCounts = await this.getProjectsCounts()
+
+    return {
+      combined: this.toCombinedActivity(projectsCounts),
+      projects: this.toProjectsActivity(projectsCounts),
+    }
+  }
+
+  private async getProjectsCounts(): Promise<ProjectsCounts> {
+    const projectPromises = this.projects
+      .filter((p) => !!p.transactionApi)
+      .map(async (p) => {
+        const repository =
+          p.transactionApi?.type === 'rpc'
```

Comment:
This will silently fail once we add support for zkSync. I think a `switch` statement with explicit failure on default clause would be much better.

Reply:
Propobably a helper function would help with redability too...